ironfish/src/rpc/routes/eth/getTransactionByHash.ts

```diff
@@ -0,0 +1,143 @@
+      yParity: yup.string().defined(),
+    })
+    .defined()
+
+registerEthRoute<typeof GetTransactionByHashRequestSchema, GetTransactionByHashResponse>(
+  `eth_getTransactionByHash`,
+  `${ApiNamespace.eth}/getTransactionByHash`,
+  GetTransactionByHashRequestSchema,
+  async (request, node): Promise<void> => {
+    Assert.isInstanceOf(node, FullNode)
+    const transactionHash =
+      await node.chain.blockchainDb.getEthTransactionHashToTransactionHash(
+        Buffer.from(EthUtils.remove0x(request.data), 'hex'),
+      )
+    if (!transactionHash) {
+      throw new RpcNotFoundError(`Transaction ${request.data} not found`)
+    }
+    const blockHash = await node.chain.getBlockHashByTransactionHash(transactionHash)
+
+    if (!blockHash) {
+      throw new RpcNotFoundError(`Block not found for transaction ${request.data}`)
+    }
+    const blockHeader = await node.chain.getHeader(blockHash)
+    if (!blockHeader) {
+      throw new RpcNotFoundError(`Block header not found for transaction ${request.data}`)
+    }
+    const transactions = await node.chain.getBlockTransactions(blockHeader)
+
+    let transaction:
+      | {
+          transaction: Transaction
+          initialNoteIndex: number
+          sequence: number
+          blockHash: Buffer
+          previousBlockHash: Buffer
+          timestamp: Date
+        }
+      | undefined = undefined
+    let transactionNum = 0
+    for (const tx of transactions) {
+      if (tx.transaction.hash().equals(transactionHash)) {
+        transaction = tx
+        break
+      }
+      transactionNum += 1
+    }
+    if (!transaction) {
+      throw new RpcNotFoundError(`Transaction ${request.data} not found`)
+    }
+    if (!transaction.transaction.evm) {
+      throw new RpcNotFoundError(`Transaction ${request.data} does not have EVM description`)
+    }
+    const ethTransaction = evmDescriptionToLegacyTransaction(transaction.transaction.evm)
+    // TODO deal with items that are mocked
+    request.end({
+      blockHash: EthUtils.prefix0x(transaction.blockHash.toString('hex')),
+      blockNumber: EthUtils.prefix0x(transaction.sequence.toString(16)),
+      transactionIndex: EthUtils.prefix0x(transactionNum.toString(16)),
+      from: ethTransaction.getSenderAddress().toString(),
+      gas: EthUtils.prefix0x(ethTransaction.gasLimit.toString(16)),
+      gasPrice: EthUtils.prefix0x(ethTransaction.gasPrice.toString(16)),
+      maxFeePerGas: '0x',
+      maxPriorityFeePerGas: '0x',
+      hash: EthUtils.prefix0x(Buffer.from(ethTransaction.hash()).toString('hex')),
+      input: EthUtils.prefix0x(Buffer.from(ethTransaction.data).toString('hex')),
+      nonce: EthUtils.prefix0x(ethTransaction.nonce.toString(16)),
+      to: ethTransaction.to === undefined ? null : ethTransaction.to.toString(),
+      value: EthUtils.prefix0x(ethTransaction.value.toString(16)),
+      type: EthUtils.prefix0x(ethTransaction.type.toString(16)),
+      accessList: [],
+      chainId: '0x42069',
```

Comment:
😂 

at some point we might need to figure out what to use for the chainId. I think it should ideally be a parameter to signing, too